mtags/src/main/scala-3/scala/meta/internal/pc/Completions.scala

```diff
@@ -0,0 +1,104 @@
+package scala.meta.internal.pc
+
+import dotty.tools.dotc.ast.tpd.{Ident, Apply, Tree, NamedArg, Template, Block, Literal, Select}
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.util.SourcePosition
+import dotty.tools.dotc.core.Constants.Constant
+import dotty.tools.dotc.interactive.Completion
+import dotty.tools.dotc.core.Symbols.Symbol
+import dotty.tools.dotc.core.Names.Name
+
+trait Completions {
+  def completionPosition(
+    pos: SourcePosition,
+    path: List[Tree],
+  )(using ctx: Context): List[Completion] = {
+    path match {
+      case (ident: Ident) :: (app: Apply) :: _ => // fun(arg@@)
+        ArgCompletion(Some(ident), app, pos).contribute
+      case (app: Apply) :: _ => // fun(@@)
+        ArgCompletion(None, app, pos).contribute
+      case _ =>
+        Nil
+    }
+  }
+}
+
+case class ArgCompletion(
+  ident: Option[Ident],
+  apply: Apply,
+  pos: SourcePosition,
+)(using ctx: Context) {
+  val method = apply.fun
+  val methodSym = method.symbol
+
+  // paramSymss contains both type params and value params
+  val vparamss = methodSym.paramSymss.filter(
+    params => params.forall(p => p.isTerm)
+  )
+  val argss = collectArgss(apply)
+
+  // get params and args we are interested in
+  // e.g.
+  // in the following case, the interesting args and params are 
+  // - params: [apple, banana]
+  // - args: [apple, b]
+  // ```
+  // def curry(x; Int)(apple: String, banana: String) = ???
+  // curry(1)(apple = "test", b@@)
+  // ```
+  val (baseParams, baseArgs) = vparamss.zip(argss).lastOption.getOrElse((Nil, Nil))
+
+  val args = ident
+    .map(i => baseArgs.filterNot(_ == i)).getOrElse(baseArgs)
+    .filterNot(isUselessLiteral)
+
+  val isNamed: Set[Name] = args.iterator
+    .zip(baseParams.iterator)
+    // filter out synthesized default args
+    .filterNot {
+      case (Ident(name), _) => name.toString.contains("$")
```

Comment:
I wonder if we should indeed do this? Maybe Scala 3 works differently here? This will also cause people to not be able to use named params that contain `$`, which I think is a bug and is reported under https://github.com/scalameta/metals/issues/2348

Reply:
Yeah, in e745df8 tried no to use `$` for filtering out synthetic params and arg getters.

- we could filter out synthesized param like `evidence$1` for something like `def foo[T: Ordering](...)`.
  - https://github.com/scalameta/metals/pull/2369/commits/e745df83da8341f420fd8b176d226ca4509393b9#diff-020d74d58479b87b103a4b80f6ac22e6dd510390fa221baad3305cba2b321649R75
- and we could filter out default args by using `DefaultGetterName` instead of just checking name contains `$` or not.
  - https://github.com/lampepfl/dotty/blob/a5d3ba96b4fd9bcefe3d6d1e7ef69453fe1c0fee/compiler/src/dotty/tools/dotc/core/NameKinds.scala#L328

Maybe we could filter out synthesized params by using `Symbol#isStynthetic` in scala2 :)